mage_integrations/mage_integrations/sources/api/__init__.py

```diff
@@ -86,12 +94,71 @@ def discover(self, streams: List[str] = None) -> Catalog:
 
         return Catalog(streams)
 
-    def __build_response(self):
+    def _deal_with_google_sheets(self, response, separator, header):
+        url = self.config['url']
+
+        if url.endswith('output=csv'):
+            df = polars.read_csv(StringIO(response.content.decode()), separator=separator,
+                                 has_header=header).to_pandas()
+            return df
+        elif url.endswith('output=tsv'):
+            df = polars.read_csv(StringIO(response.content.decode()), separator='\t',
+                                 has_header=header).to_pandas()
+            return df
+        elif url.endswith('output=xlsx'):
+            df = polars.read_excel(BytesIO(response.content),
+                                   read_csv_options={"separator": separator, "has_header": header})\
+                                   .to_pandas()
+            return df
+        else:
+            raise Exception('Extension not allowed, please use CSV,TSV or XLSX')
+
+    def _check_response_type(self, response):
```

Comment:
The purpose of this method is to not only check response type, right? You can also parse the response to polars dataframe for some cases.
Could you separate the logic of check response type and parse the response?

Reply:
_check_response_type now only returns mime_type or google_sheets  ( 287c8da3404a845397c1e58290fe23ce15f0a119 )